### PR TITLE
Use libelf and libdwarf from conda

### PR DESCRIPTION
**Pull request by timsnyder-siv**
Changes dependencies for libelf and libdwarf to come from conda packages instead of submodules that are compiled
from source.

#### UI / API Impact

Needs to update driver to newer APIs of libelf and libdwarf but shouldn't change firesim-user facing API/UI 

#### Verilog / AGFI Compatibility

Doesn't touch Verilog or AGFI.

### Contributor Checklist
- [X] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [X] Did you add Scaladoc/docstring/doxygen to every public function/method?
- [X] Did you add at least one test demonstrating the PR?
- [X] Did you delete any extraneous prints/debugging code?
- [X] Did you state the UI / API impact?
- [X] Did you specify the Verilog / AGFI compatibility impact?

- [X] If applicable, did you regenerate and publicly share default AGFIs?

- [X] If applicable, did you apply the `ci:fpga-deploy` label?

- [X] If applicable, did you apply the `Please Backport` label?

### Reviewer Checklist (only modified by reviewer)
- [ ] Is the title suitable for inclusion in the changelog and does the PR have a `changelog:<topic>` label?
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

**Comment by timsnyder-siv** [comment]
@a0u and @sagark Here's a PR that bumps the versions of libelf and libdwarf to latest available via conda-forge.  The CI will fail with API differences in libdwarf.  I wanted to go ahead and push this PR.  If you guys have time to get things building again with these versions, it would be helpful.

It might be good for us to add a short test that exercises the libelf and libdwarf portions of TracerV as a part of `ci:fpga_deploy` tests.

We talked about this briefly back in https://github.com/firesim/firesim/pull/986#discussion_r846552592

**Comment by timsnyder-siv** [comment]
Looks like there are a couple of version bumps of libdwarf that I haven't noticed.   I'll look at why 0.4.1 is failing in CI at https://github.com/conda-forge/libdwarf-feedstock/pull/2 and get it merged.

File: scripts/machine-launch-script.sh
Comment (abejgonzalez) [inline scripts/machine-launch-script.sh:169]: @timsnyder-siv Do you see any issues with making this a bit more flexible?

File: scripts/machine-launch-script.sh
Comment (abejgonzalez) [inline scripts/machine-launch-script.sh:286]: This now matches the current `libdwarf` version that we use.

File: scripts/machine-launch-script.sh
Comment (timsnyder-siv) [reply scripts/machine-launch-script.sh:169]: Since we're installing 'miniforge3' and it doesn't have the 'Defaults' channel that has known conflicts with conda-forge, this should be okay.  It does make it much more complicated to determine which channel any given package is going to come from but I see why you need to have this with the libdwarf-dev

**Comment by timsnyder-siv** [comment]
@abejgonzalez how is the `/scratch/buildbot/conda/envs/ci/include` maintained on `firesim1`?  It seems like the driver build is failing to find libdwarf.h there.

File: scripts/machine-launch-script.sh
Comment (timsnyder-siv) [reply scripts/machine-launch-script.sh:169]: @abejgonzalez see also my comment on the chipyard conda PR at https://github.com/ucb-bar/chipyard/pull/1163/files#r953157747. Since you guys are just saying 'install conda' and point to the Anaconda Inc instructions, this loosening might be problematic for anyone who also has 'defaults' in their conda configuration

**Comment by abejgonzalez** [comment]
> @abejgonzalez how is the `/scratch/buildbot/conda/envs/ci/include` maintained on `firesim1`? It seems like the driver build is failing to find libdwarf.h there.

Manually. I will add a PR to programmatically update the env when I get the chance. Until then I have to update the environment myself. If this PR looks sufficiently fine (which judging by your questions seems fine), then I'll update the CI. Or we can hold off on this until I update the CI scripts.

**Comment by abejgonzalez** [comment]
@mergifyio rebase

**Comment by abejgonzalez** [comment]
> @abejgonzalez how is the `/scratch/buildbot/conda/envs/ci/include` maintained on `firesim1`? It seems like the driver build is failing to find libdwarf.h there.

I added these deps to the `ci` conda environment. This manual intervention should matter anymore after #1180

**Comment by abejgonzalez** [comment]
@mergifyio rebase

**Review by abejgonzalez — APPROVED** [review]
LGTM.

File: scripts/machine-launch-script.sh
Comment (abejgonzalez) [reply scripts/machine-launch-script.sh:169]: I've addressed this in the CY PR you mentioned. In any case, #1180 will double-check this by adding the `nodefaults` channel to the `conda-requirements.yaml` file.

**Comment by abejgonzalez** [comment]
@timsnyder-siv I LGTM'ed your code, if you give me an LGTM on my changes then I'll go ahead and merge this.

**Comment by timsnyder-siv** [comment]
You might want to double check that the order of the channels is what you want but looks good to merge

**Comment by abejgonzalez** [comment]
It seems fine to me. I think the future PRs will double-check things as well. Merging

**Pull request merged**
